src/App.jsx

```diff
@@ -1,10 +1,106 @@
+import axios from 'axios'
 import './App.css';
+import {useState} from "react";
+import globe from "./assets/spinning-globe.jpg"
+import worldmap from "./assets/world_map.png"
+import regionColorNames from "./helpers/regionColorHelper.js";
+import sortPopulationHelper from "./helpers/sortPopulationHelper.js";
 
 function App() {
 
+    const [countries, setCountries] = useState("")
+    const [countryName, setCountryName] = useState("")
+    const [countryData, setCountryData] = useState("")
+    const [error, setError] = useState("")
+
+    async function fetchAllCountries() {
+        try {
+            const result = await axios.get('https://restcountries.com/v3.1/all');
+            console.log(result.data)
+            const sortedCountries = sortPopulationHelper(result.data)
+            setCountries(sortedCountries);
+        } catch (e) {
+            console.error(e);
+        }
+    }
+
+    async function fetchCountry() {
+        try {
+            setError("")
+
+            const result = await axios.get(`https://restcountries.com/v3.1/name/${countryName}`);
+            console.log(result.data)
+            setCountryData(result.data)
+            setCountryName("");
+        } catch (e) {
+            console.error(e)
+            setError(`${countryName} bestaat niet. Probeer het opnieuw.`)
+        }
+    }
+
     return (
         <>
-            Maak je applicatie hier!
+            <main className="page-container">
+                <section className="all-countries-section">
+                <span className="header-image-wrapper">
+                    <img className="header-image" src={worldmap} alt="world-map"/>
+                </span>
+                        <button className="all-countries-button" type="button" onClick={fetchAllCountries}>Zoek alle landen</button>
+                    <ul className="countries-container">
+                        {countries && countries.map((country) => (
```

Comment:
```suggestion
                        {countries.map((country) => (
```
(Mits de default waarde `[]` is)